### ISSUE TITLE
[v 0.9.2] Recent changes to label ordering make errors more confusing

### ISSUE THREAD

**Issue by tnowacki** [brendanzab/codespan#231]
I find the recent change to the order in which labels displayed make a lot of error messages *really* confusing to read when displayed in the new format. 
https://github.com/brendanzab/codespan/commit/c3dd8465c762d7ead9471426d56fa539e4b324a2

For example, I find 
```
    ┌── tests/move_check/borrows/assign_local_combo_invalid.move:14:9 ───
    │
 14 │         s = S { f: 0, g: 0 };
    │         ^ Invalid assignment of local 's'
    ·
 13 │         if (cond) f = &s.f else f = &s.g;
    │                                     ---- It is still being borrowed by this reference
    │
```
A lot easier to reason through than the new version:
```
    ┌── tests/move_check/borrows/assign_local_combo_invalid.move:13:37 ───
    │
 13 │         if (cond) f = &s.f else f = &s.g;
    │                                     ---- It is still being borrowed by this reference
 14 │         s = S { f: 0, g: 0 };
    │         ^ Invalid assignment of local 's'
    │
```

Sometimes, if they are on the same line, it gets really really weird:
```
   ┌── tests/move_check/expansion/global_access_pack.move:3:13 ───
   │
 3 │         ::S { }
   │             ^ Unexpected '{'
   ·
 3 │         ::S { }
   │             - Expected '('
   │
```
Becomes
```
   ┌── tests/move_check/expansion/global_access_pack.move:3:13 ───
   │
 3 │         ::S { }
   │             - Expected '('
   │             ^ Unexpected '{'
   │
```
Do you have any suggests to how to fix this if we are on version 0.9.2?

Further context and discussion:
https://github.com/libra/libra/pull/3447

**Comment by jonathandturner** [comment]
@tnowacki - I'm not sure about the multiple labels on the same line, but we talk a bit about the ordering of lines in the original design for rustc - https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html

I wouldn't say there's like a perfect solution here, but being able to see the code in the order you wrote it keeps you from having to look left multiple times to find the line numbers. It's a small time save, though I'd be curious to see if there were better ways to shave off a little time there.

**Comment by brendanzab** [comment]
Yeah, I think we could do a better job of preserving the ordering of labels on the same line, but yeah, I think that it's better to mirror the ordering of the existing source code to reduce spatial confusion. Would it be possible to reword the labels so that they are less spatially dependent, like how Rustc does it?

**Comment by tnowacki** [comment]
@jonathandturner Thanks for the link to the Rust blog! It will take some work to restate errors in way that is more "observational" than my current setup where it is imperative. 

I think the ordering of labels might be helpful for labels on the same line, but otherwise I think it will be worthwhile to go back and rework my error messages. I can definitely see the benefit of trying to display code as it was written with added annotations.

**Comment by tnowacki** [comment]
Feel free to close the issue, or keep it open if you want for the single line case

**Comment by brendanzab** [comment]
@tnowacki Thanks again for the issue - really sorry for the churn! One option could be to use version `=0.9.1`, which would mean you could merge the API updates in the mean-time.

**Comment by brendanzab** [comment]
The single line case should be fixed in #232!
